### PR TITLE
Move blueprint provided packages to peerDependencies

### LINKED ISSUE #899: Audit peerDependencies
Pulled out of #894 where `devDependencies` have been added to ~~`@babel/runtime` and~~ `@ember/string`.

As suggested by @jelhan 

>... may need to go into peerDependency to avoid a deprecation in Ember 4.10+. Adding as peer dependency would be a breaking change. Maybe better done in separate PR.

> When doing a new major for that, we should also revisit other dependencies and see if they should be peer dependencies instead. E.g. @glimmer/tracking and @glimmer/component.

### PR DESCRIPTION
**Pull request by gilest**
Experiment to see how we might fix #899

Being aggressive here and moving as much as possible from `dependencies` to `devDependencies` in an attempt to batch breaking changes and avoid another major.

### CODE REVIEW

File: ember-file-upload/package.json [hunk 1]
```diff
@@ -99,12 +98,18 @@
...
   "peerDependencies": {
+    "@ember/test-helpers": "^2.9.3",
+    "@glimmer/component": "^1.1.2",
+    "@glimmer/tracking": "^1.1.2",
     "ember-cli-mirage": "*",
-    "miragejs": "*"
+    "ember-modifier": "^3.2.7 || ^4.1.0",
+    "miragejs": "*",
+    "tracked-built-ins": "^3.1.1"
   },
```
Comment (gilest) [inline ember-file-upload/package.json:113]: Everything I've moved to `peerDependencies` is in the modern `ember-cli` blueprints.

File: ember-file-upload/package.json [hunk 2]
```diff
@@ -46,15 +46,10 @@
...
     "@ember/test-waiters": "^3.0.0",
     "@embroider/addon-shim": "^1.5.0",
     "@embroider/macros": "^1.0.0",
-    "@glimmer/component": "^1.0.4",
-    "@glimmer/tracking": "^1.0.4",
-    "ember-auto-import": "^2.0.0",
-    "ember-modifier": "^3.2.7 || ^4.0.0",
-    "tracked-built-ins": "^3.0.0"
+    "ember-auto-import": "^2.0.0"
   },
```
Comment (gilest) [inline ember-file-upload/package.json:53]: Believe the remaining dependencies are required. Keen for others to check this, though.

Comment (jelhan) [reply ember-file-upload/package.json:53]: Looks good to.

**Review by jelhan — APPROVED** [review]
Looks good to me.

**Pull request merged**